src/modules/endpoint/reducers/morphContentLanguage.js

```diff
@@ -1,83 +1,83 @@
 // @flow
 
-import { defaultRouteState } from '../../app/StateType'
+import type { CategoryRouteStateType, CityContentStateType, EventRouteStateType } from '../../app/StateType'
 import { mapValues } from 'lodash/object'
-import { reduce } from 'lodash/collection'
-import { CategoryModel } from '@integreat-app/integreat-api-client'
+import { EventModel } from '@integreat-app/integreat-api-client'
 import type { MorphContentLanguageActionType } from '../../app/StoreActionType'
-import type {
-  CategoryRouteStateType,
-  CityContentStateType,
-  EventRouteStateType
-} from '../../app/StateType'
-
-const translatePath = (model: CategoryModel, currentCity: string, newLanguage: string) => {
-  if (model.isRoot()) {
-    return `/${currentCity}/${newLanguage}`
-  } else {
-    return model.availableLanguages.get(newLanguage)
-  }
-}
-
-const translateChildren = (models, newCategoriesMap, children, currentCity, newLanguage) =>
-  reduce(children, (result, value: Array<string>, path: string) => {
-    const translatedKey = translatePath(models[path], currentCity, newLanguage)
+import CategoriesMapModel from '@integreat-app/integreat-api-client/models/CategoriesMapModel'
+import forEachTreeNode from '../../common/forEachTreeNode'
+import { isEmpty } from 'lodash'
 
-    if (!translatedKey) {
-      // TODO: This is code for debugging which could help in the future. Remove once this has been tested in NATIVE-116
-      console.warn(`Path ${path} is not translatable!`)
-      return result
-    }
+const categoryRouteTranslator = (newCategoriesMap: CategoriesMapModel, city: string, newLanguage: string) =>
```

Comment:
I think city is not required here, is it?

Reply:
No, thanks. Removed.